samples/cellular/modem_shell/src/location/location_shell.c

```diff
@@ -282,9 +384,18 @@ void location_ctrl_event_handler(const struct location_event_data *event_data)
 		location_print_data_details(event_data->method, &event_data->location.details);
 #endif
 		if (gnss_location_to_cloud) {
+			gnss_location_work_data.timestamp_ms = ts_ms;
 			gnss_location_work_data.loc_evt_data = *event_data;
 			gnss_location_work_data.format = gnss_location_to_cloud_format;
+#if defined(CONFIG_LOCATION_DATA_DETAILS)
+			/* Cannot use the same workq as metrics does, using system workq */
+			/* TODO: Should not use the system work queue either, because this can
+			 * block it for too long. This should be refactored at some point.
+			 */
+			k_work_submit(&gnss_location_work_data.work);
+#else
 			k_work_submit_to_queue(&mosh_common_work_q, &gnss_location_work_data.work);
+#endif
```

Comment:
When location is retrieved quickly after another one so that the previous position/details have not been sent to cloud, MOSH will currently overwrite previous data so we'd miss some of the data.

These are in different work queue to be able to execute them faster. My plan of putting these to same worker (and hopefully same cloud message) is not particularly good from this perspective.

If interval is small like one second and sending takes several seconds, there is no solution to this as any buffering won't help as the buffer will eventually get full and we need to drop/miss some data.

I suggest we add an error print when the worker is still running when data would be overwritten.
And I'd like to send these in the same worker to make things simpler.

Reply:
> (and hopefully same cloud message)

Everything else from this comment is done except this.

Reply:
> > (and hopefully same cloud message)
> 
> Everything else from this comment is done except this.

I don't think we'll try to combine these into the same cloud message. Not really worth the effort at this point.